### PR TITLE
Fix issue #1814

### PR DESCRIPTION
**Pull request by M374LX**
The problem described by issue #1814 was caused by combo box values being read from the settings loaded at startup, rather than from the settings saved to be loaded when the program restarts.

### CODE REVIEW

**Comment by tresf** [comment]
Thanks @M374LX!

@badosu can you help review this PR?  I'd like to make sure there aren't any downsides to this approach. :+1: 

-Tres

File: src/gui/dialogs/SetupDialog.cpp [hunk 1]
```diff
@@ -0,0 +1 @@
+// Please ignore this file
```
Comment (badosu) [inline src/gui/dialogs/SetupDialog.cpp]: Why is this file needed at all?

Comment (M374LX) [inline src/gui/dialogs/SetupDialog.cpp]: This was actually a failed attempt to move SetupDialog.cpp to the "dialogs" directory. The move would require changes in CMake files, which might be a better idea to do in a distinct pull request.

**Comment by badosu** [comment]
@M374LX Really nice fix, thanks for contributing!

From what I see we were loading a harcoded value instead of the one that you get from `ConfigManager` right?

Would you be able to delete that additional line on `SetupDialog.cpp` and squash all your commits into one?

Since this should be a simple atomic change one commit should suffice.

See how to use `git rebase -i` to squash your commits here: https://help.github.com/articles/about-git-rebase/.

If you have any questions let me know, I usually hang out on #lmms on freenode, so you can chat with me directly there.

**Comment by M374LX** [comment]
@badosu 
Do you refer to the `SetupDialog.cpp` file inside the `dialogs` folder, which contains a commented line saying "Please ignore this file"?

**Comment by badosu** [comment]
@M374LX Yep, that change is quite irrelevant to the issue you solved.

If you want to fix this now, in a different commit that would be nice, though.

**Comment by badosu** [comment]
@tresf @M374LX As far as code is concerned this PR looks fine!

@curlymorphic Reminded me of how git can be annoying sometimes, so if this is a problem for @M374LX let use know that we can proceed for the merge.

**Comment by tresf** [comment]
I agree with the squashing of commits (small change, no need for 4 separate commits).  Once this is done, we'll merge it into master branch.  Just post here once it has been completed because GitHub does not send notifications for additional commits.  Thanks!

**Comment by badosu** [comment]
@M374LX it looks like the references I shared are not very good, these ones are better for your issue:

- https://ariejan.net/2011/07/05/git-squash-your-latests-commits-into-one/
- http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html

Don't worry about some warnings, since your branch is only yours you're free to change it the way you want, just pay attention to some details as you're doing a somewhat delicate operation.

Do not forget to `git push -f` on the end.

**Comment by M374LX** [comment]
@tresf @badosu
Done?

**Comment by badosu** [comment]
:+1:

**Comment by badosu** [comment]
@tresf This issue may have solved the issue when writing the file, but the previous usage was quite decent, it checked what backend LMMS was **actually** using instead of the written on file.

Although I agree with the decision of showing the one on the file we should keep this is mind.

**Comment by tresf** [comment]
This was my exact concern.  Thanks.  We'll make mental note.
